editor/selectors.js

```diff
@@ -168,19 +168,39 @@ export function getEditedPostVisibility( state ) {
 }
 
 /**
- * Return true if the post being edited has already been published.
+ * Return true if the current post has already been published.
  *
  * @param  {Object}   state Global application state
  * @return {Boolearn}       Whether the post has been bublished
  */
-export function isEditedPostPublished( state ) {
+export function isCurrentPostPublished( state ) {
 	const post = getCurrentPost( state );
 
 	return [ 'publish', 'private' ].indexOf( post.status ) !== -1 ||
 		( post.status === 'future' && moment( post.date ).isBefore( moment() ) );
 }
 
 /**
+ * Return true if the post being edited has already been published.
+ *
+ * @param  {Object}   state Global application state
+ * @return {Boolearn}       Whether the post has been bublished
+ */
+export function isEditedPostPublished( state ) {
```

Comment:
If I recall correctly, the issue is that if the user chooses a date in the past for their draft post, the edited representation of the post becomes "published", but we can't necessarily use the published autosave flow because the saved post is still a draft.

... I think there's still something amiss here though, because in the `AUTOSAVE` effect above I'm still testing against the edited version of the post, not the "current" (saved) copy.